From bb246f368cfb99738086164bbe66fd0117d42d28 Mon Sep 17 00:00:00 2001
From: Chenghua Xu <xuchenghua@loongson.cn>
Date: Thu, 13 Oct 2016 10:43:00 +0800
Subject: [PATCH 4/7] Fix ld default set isa flags.

    Set ld default isa flags to loongson3a.
    bdf/
    * elfxx-mips.c: Fixed.
---
 bfd/elfxx-mips.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/bfd/elfxx-mips.c b/bfd/elfxx-mips.c
index 6a897b70d6..2a984a4a6f 100644
--- a/bfd/elfxx-mips.c
+++ b/bfd/elfxx-mips.c
@@ -11912,6 +11912,10 @@ mips_set_isa_flags (bfd *abfd)
   switch (bfd_get_mach (abfd))
     {
     default:
+    case bfd_mach_mips_loongson_3a:
+      val = E_MIPS_ARCH_64R2 | E_MIPS_MACH_LS3A;
+      break;
+
     case bfd_mach_mips3000:
       val = E_MIPS_ARCH_1;
       break;
@@ -11993,10 +11997,6 @@ mips_set_isa_flags (bfd *abfd)
       val = E_MIPS_ARCH_64 | E_MIPS_MACH_SB1;
       break;
 
-    case bfd_mach_mips_loongson_3a:
-      val = E_MIPS_ARCH_64R2 | E_MIPS_MACH_LS3A;
-      break;
-
     case bfd_mach_mips_octeon:
     case bfd_mach_mips_octeonp:
       val = E_MIPS_ARCH_64R2 | E_MIPS_MACH_OCTEON;
-- 
2.16.1

